### PR TITLE
Refactor provision-mbl tests using pytest

### PR DESCRIPTION
**Pull request by diegorusso**

### CODE REVIEW

File: ci/lava/tests/test-provision-mbl.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            ]
+        )
+        if exit_code:
+            print(stdout)
+            print(stderr)
+        assert exit_code == 0
+
+    def test_reject_invalid_key(self, execute_helper):
+        """Test invalid API key saving using mbl-cli."""
+        exit_code, stdout, stderr = execute_helper.execute_command(
```
Comment (petedyerarm) [inline ci/lava/tests/test-provision-mbl.py]:
```suggestion
        exit_code, stdout, stderr = execute_helper.send_mbl_cli_command(
            ["save-api-key", "invalid_key"], TestProvisionMbl.dut_addr
```

File: ci/lava/tests/test-provision-mbl.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            ["mbl-cli", "save-api-key", "invalid_key"]
```
Comment (petedyerarm) [inline ci/lava/tests/test-provision-mbl.py]:
```suggestion
```

File: ci/lava/tests/test-provision-mbl.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        current_dir = Path().cwd()
+        TestProvisionMbl.certificate = "".join(current_dir.parts[1])
+
+        # Create a working directory for the following tests
+        directory = "/tmp/update-resources"
+        if not os.path.exists(directory):
+            os.mkdir(directory)
+        os.chdir(directory)
+
+    def test_manifest_ini(self, execute_helper):
```
Comment (rwalton-arm) [inline ci/lava/tests/test-provision-mbl.py]: This name is confusing, why isn't this called `test_manifest_init`?

Comment (diegorusso) [reply ci/lava/tests/test-provision-mbl.py]: Fixed: 4be8f3e7d0a7fe405081c27aa914551079d59977

**Pull request merged**